KPCTabsControl/TabsControl.swift

```diff
@@ -461,12 +464,24 @@ public class TabsControl: NSControl, TabEditingDelegate {
      - parameter index: An integer indicating the index of the item to be selected.
      */
     public func selectItemAtIndex(index: Int) {
-        let buttons = self.tabButtons()
-        if buttons.count > index {
-            self.selectTab(buttons[index])
+        guard let button = self.tabButtons()[safe: index] else { return }
+
+        self.selectTab(button)
+    }
+
+    private func updateButtonStatesForSelection() {
+        for button in self.tabButtons() {
+            if let equatableSelectedItem = selectedItem as? String,
```

Comment:
We may have a problem here: Swift bridges stuff like `String` to `AnyObject` via `NSString` perhaps, but under the hood the equality checks fail. It's not very promising to support every struct from Foundation here. Casting to `Equatable` doesn't work, either.

My suggestion: introduce a `TabButtonItem` protocol which sports an equality check. People who use strings can extend the `String` type on the client side.